Example/Example/EditorDemoController.swift

```diff
@@ -591,25 +608,62 @@ extension EditorDemoController : Aztec.FormatBarDelegate {
+                                                   fromBarItem barItem: FormatBarItem,
+                                                   selectedRowIndex index: Int?,
+                                                   onSelect: OptionsTableViewController.OnSelectHandler?) {
         // Hide the input view if we're already showing these options
         if let optionsViewController = optionsViewController,
             optionsViewController.options == options {
+            if presentedViewController != nil {
+              dismiss(animated: true, completion: nil)
+            }
+
             self.optionsViewController = nil
             changeRichTextInputView(to: nil)
             return
         }
 
         optionsViewController = OptionsTableViewController(options: options)
         optionsViewController.cellDeselectedTintColor = .gray
-        optionsViewController.onSelect = onSelect
+        optionsViewController.onSelect = { [weak self] selected in
+            if self?.presentedViewController != nil {
+                self?.dismiss(animated: true, completion: nil)
+            }
+
+            onSelect?(selected)
+        }
+
+        let selectRow = {
+            if let index = index {
+                self.optionsViewController.selectRow(at: index)
+            }
+        }
+
+        if UIDevice.current.userInterfaceIdiom == .pad  {
+            optionsViewController.modalPresentationStyle = .popover
+            optionsViewController.popoverPresentationController?.permittedArrowDirections = [.down]
+            optionsViewController.popoverPresentationController?.sourceView = view
+
+            let frame = barItem.superview?.convert(barItem.frame, to: UIScreen.main.coordinateSpace)
 
-        self.addChildViewController(optionsViewController)
-        changeRichTextInputView(to: optionsViewController.view)
-        optionsViewController.didMove(toParentViewController: self)
+            optionsViewController.popoverPresentationController?.sourceRect = view.convert(frame!, from: UIScreen.main.coordinateSpace)
+            optionsViewController.popoverPresentationController?.backgroundColor = .white
+            optionsViewController.popoverPresentationController?.delegate = self
 
-        if let index = index {
-            optionsViewController.selectRow(at: index)
+            if presentedViewController != nil {
+                dismiss(animated: true, completion: { 
+                    self.present(self.optionsViewController, animated: true, completion: selectRow)
+                })
+            } else {
+                present(optionsViewController, animated: true, completion: selectRow)
+            }
+
+        } else {
+            self.addChildViewController(optionsViewController)
+            changeRichTextInputView(to: optionsViewController.view)
+            optionsViewController.didMove(toParentViewController: self)
+            selectRow()
         }
```

Comment:
Can we split this method into a couple smaller? Couple ideas below!

A. Show Options could call one of these two in the end:
```
- displayOptionsOnPad(optionsViewController:...
- displayOptionsOnPhone
```

B. potentially a method just dedicated to instantiating the Options VC?